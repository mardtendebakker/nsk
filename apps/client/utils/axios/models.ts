export interface BaseCustomer {
  id?: number,
  name?: string,
  representative?: string,
  email?: string,
  phone?: string,
  phone2?: string,
  street?: string,
  street_extra?: string,
  city?: string,
  country?: string,
  state?: string,
  zip?: string,
  street2?: string,
  street_extra2?: string,
  city2?: string,
  country2?: string,
  state2?: string,
  zip2?: string,
}

export interface Customer extends BaseCustomer {
  kvk_nr?: string,
  is_partner?: number,
}

export interface Company {
  id: string,
  name: string,
  acompany?: { id: string, name: string }
}

export interface Order {
  id?: number,
  order_nr?: string
  remarks?: string,
  order_date: string,
  discount?: number,
  transport?: number,
  is_gift?: boolean,
  delivery_instructions?: string
  order_status: { name: string, color: string }
  acompany_aorder_supplier_idToacompany?: Company,
  acompany_aorder_customer_idToacompany?: Company
}

export interface AttributeOption {
  id: number,
  name: string,
  price: number
}

export interface Attribute {
  id?: number,
  name: string,
  type: number,
  options: AttributeOption[]
}

export interface ProductType {
  id?: number,
  name: string,
  attributes: Attribute[]
}

export interface Task {
  name: string,
  description: string,
  status: number
}

export interface Product {
  id?: number,
  hold: number,
  location: string,
  name: string,
  order_date: string,
  order_nr: string,
  price: number,
  purch: number,
  sale: number,
  sku: string,
  sold: number,
  splittable: boolean,
  stock: number,
  tasks: Task[],
  updated_at: string
}

export interface Supplier extends BaseCustomer {
  partner?: string,
}

export interface User {
  id?: string,
}
