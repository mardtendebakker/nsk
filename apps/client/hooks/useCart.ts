import { useEffect, useState } from 'react';
import cartStore, {
  ADD_MODULE_REQUEST,
  CLEAR_CART_REQUEST,
  Cart, EVENTS, Module, REMOVE_MODULE_REQUEST,
} from '../stores/cart';

const useCart = (): {
  state: Cart,
  removeModule:(module: Module)=> void,
  addModule:(module: Module)=> void,
  clearCart:()=> void,
  totalModulesCount: () => number
  totalAmount: () => number
} => {
  const [state, setState] = useState<Cart>(cartStore.cart);

  function cb() {
    setState(cartStore.cart);
  }

  useEffect(() => {
    EVENTS.forEach((event) => {
      cartStore.on(event, cb);
    });

    return () => {
      EVENTS.forEach((event) => {
        cartStore.removeListener(event, cb);
      });
    };
  }, []);

  return {
    state,
    removeModule: (module: Module) => {
      cartStore.emit(REMOVE_MODULE_REQUEST, module);
    },
    addModule: (module: Module) => {
      cartStore.emit(ADD_MODULE_REQUEST, module);
    },
    clearCart: () => {
      cartStore.emit(CLEAR_CART_REQUEST);
    },
    totalModulesCount: () => state.modules.length,
    totalAmount: () => {
      let total = 0;

      state.modules.forEach(({ price }) => {
        total += price;
      });

      return total;
    },
  };
};

export default useCart;
