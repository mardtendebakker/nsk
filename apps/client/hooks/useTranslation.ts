import { useEffect, useState } from 'react';
import { trans, setConfig } from 'itranslator';
import { setDefaultOptions } from 'date-fns';
import {
  nl, enUS, de, fr, ar, faIR,
} from 'date-fns/locale';
import Config from 'itranslator/lib/type/config';
import { getDefaultLocale, setDefaultLocale } from '../utils/storage';
import EventEmitter from '../utils/eventEmitter';
import nlSource from '../public/translations/nl';
import enSource from '../public/translations/en';
import deSource from '../public/translations/de';
import frSource from '../public/translations/fr';
import arSource from '../public/translations/ar';
import faSource from '../public/translations/fa';

export const LOCALE_UPDATED = 'LOCALE_UPDATED';
export type Trans = (key: string, config?: Config | undefined) => string;

export const localeMapping = {
  nl: nlSource,
  en: enSource,
  de: deSource,
  ar: arSource,
  fr: frSource,
  fa: faSource,
};

export const dateFnsMapping = {
  nl,
  en: enUS,
  de,
  fr,
  ar,
  fa: faIR,
};

class LocaleEmitter extends EventEmitter {
  #locale = getDefaultLocale() || 'nl';

  constructor() {
    super();
    setDefaultOptions({ locale: dateFnsMapping[this.#locale] });
  }

  get locale() { return this.#locale; }

  emit(event: string, payload: string) {
    if (LOCALE_UPDATED === event) {
      this.#locale = payload;

      setConfig({ source: localeMapping[payload] });
      setDefaultOptions({ locale: dateFnsMapping[payload] });
      setDefaultLocale(payload);
    }

    return super.emit(event, payload);
  }
}

export const localeStore = new LocaleEmitter();

const useTranslation = (): { updateLocale: (locale: string)=> void, trans: Trans, locale: string } => {
  const [locale, setLocale] = useState<string>(localeStore.locale);

  useEffect(() => {
    localeStore.on(LOCALE_UPDATED, setLocale);
    return () => {
      localeStore.removeListener(LOCALE_UPDATED, setLocale);
    };
  }, [setLocale]);

  const updateLocale = (newLocale: string) => {
    localeStore.emit(LOCALE_UPDATED, newLocale);
    setLocale(newLocale);
  };

  return {
    updateLocale,
    trans,
    locale,
  };
};

export default useTranslation;
