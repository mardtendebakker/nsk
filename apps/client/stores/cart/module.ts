export interface Module {
  name: string;
  price: number;
}
