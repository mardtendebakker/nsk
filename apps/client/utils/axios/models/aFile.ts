export interface AFile {
  id?: number,
  unique_server_filename?: string,
  original_client_filename?: string,
  discr?: string
}
