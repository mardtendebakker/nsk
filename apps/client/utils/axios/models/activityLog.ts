export interface ActivityLog {
  id: number;
  username?: string;
  method?: string;
  route?: string;
  params?: string;
  body?: string;
  query?: string;
  before?: string;
  model?: string;
  action?: string;
  bulk: boolean;
  createdAt: string;
  updated_at: string;
}
