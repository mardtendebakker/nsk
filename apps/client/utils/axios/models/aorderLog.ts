export interface OrderStatus {
  id: number;
  name: string;
  color: string;
}

export interface AorderLog {
  id: number;
  username: string;
  previous_status_id: number;
  status_id: number;
  previous_status: OrderStatus;
  status: OrderStatus;
  created_at: string;
  updated_at: string;
}
