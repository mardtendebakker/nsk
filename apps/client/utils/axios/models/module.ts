export interface ModuleListItem {
  id: number;
  name: string;
  active: boolean;
  price: number;
  freeTrialUsed: boolean;
  config: { [key: string]: {
    value?: string | string[],
    required?: boolean,
    type: 'string' | 'hour' | 'multiSelect' | 'password',
    options?: string[]
  } };
  activeAt?: string;
  expiresAt?: string;
}
