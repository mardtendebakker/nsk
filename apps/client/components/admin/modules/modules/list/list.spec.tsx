import { render, screen, fireEvent } from '@testing-library/react';
import List from './list';
import useTranslation from '../../../../../hooks/useTranslation';
import useCart from '../../../../../hooks/useCart';
import { ModuleListItem } from '../../../../../utils/axios/models/module';

jest.mock('../../../../../hooks/useTranslation', () => ({
  __esModule: true,
  default: jest.fn(),
}));

jest.mock('../../../../../hooks/useCart', () => ({
  __esModule: true,
  default: jest.fn(),
}));

jest.mock('../../../../tableCell', () => ({
  __esModule: true,
  default: ({ children }) => <td>{children}</td>,
}));

describe('List Component', () => {
  const mockModules: ModuleListItem[] = [
    {
      id: 1,
      name: 'Basic Module',
      price: 99.99,
      active: false,
      freeTrialUsed: false,
      config: {
        apiKey: {
          value: null,
          required: true,
          type: 'password',
        },
        schedule: {
          value: '09:00',
          required: true,
          type: 'hour',
        },
      },
      activeAt: '2024-01-01T00:00:00Z',
      expiresAt: '2025-01-01T00:00:00Z',
    },
    {
      id: 2,
      name: 'Advanced Module',
      price: 149.99,
      active: true,
      freeTrialUsed: true,
      config: {
        mode: {
          value: 'standard',
          required: true,
          type: 'string',
          options: ['standard', 'advanced', 'expert'],
        },
        notifications: {
          value: ['email', 'sms'],
          required: false,
          type: 'multiSelect',
          options: ['email', 'sms', 'push'],
        },
      },
      activeAt: '2024-02-01T00:00:00Z',
      expiresAt: null,
    },
    {
      id: 3,
      name: 'Simple Module',
      price: 49.99,
      active: false,
      freeTrialUsed: false,
      config: null,
      activeAt: null,
      expiresAt: null,
    },
  ];

  const mockTranslations = {
    name: 'Name',
    price: 'Price',
    activeAt: 'Active At',
    expiresAt: 'Expires At',
    active: 'Active',
    actions: 'Actions',
  };

  const mockAddModule = jest.fn();
  const mockOnFreeTrial = jest.fn();
  const mockOnSettings = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();

    (useTranslation as jest.Mock).mockReturnValue({
      trans: (key: string) => mockTranslations[key],
    });

    (useCart as jest.Mock).mockReturnValue({
      addModule: mockAddModule,
      state: { modules: [] },
    });
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<List modules={mockModules} disabled={false} onFreeTrial={mockOnFreeTrial} onSettings={mockOnSettings} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders table headers correctly', () => {
    render(<List modules={[]} disabled={false} onFreeTrial={mockOnFreeTrial} onSettings={mockOnSettings} />);

    Object.values(mockTranslations).forEach((header) => {
      expect(screen.getByText(header)).toBeInTheDocument();
    });
  });

  it('renders module rows with correct data', () => {
    render(<List modules={mockModules} disabled={false} onFreeTrial={mockOnFreeTrial} onSettings={mockOnSettings} />);

    mockModules.forEach((module) => {
      expect(screen.getByText(module.name)).toBeInTheDocument();
      if (module.active) {
        expect(screen.getByTestId('CheckIcon')).toBeInTheDocument();
      }
    });
  });

  it('handles modules with missing required config', () => {
    const moduleWithMissingConfig: ModuleListItem = {
      ...mockModules[0],
      config: {
        apiKey: {
          value: null,
          required: true,
          type: 'password',
        },
      },
    };

    render(<List modules={[moduleWithMissingConfig]} disabled={false} onFreeTrial={mockOnFreeTrial} onSettings={mockOnSettings} />);

    const addToCartButton = screen.getAllByRole('button')[0];
    const freeTrialButton = screen.getAllByRole('button')[1];

    expect(addToCartButton).toBeDisabled();
    expect(freeTrialButton).toBeDisabled();
  });

  it('enables actions when all required configs are set', () => {
    const moduleWithCompleteConfig: ModuleListItem = {
      ...mockModules[0],
      config: {
        apiKey: {
          value: 'secret-key',
          required: true,
          type: 'password',
        },
      },
    };

    render(<List modules={[moduleWithCompleteConfig]} disabled={false} onFreeTrial={mockOnFreeTrial} onSettings={mockOnSettings} />);

    const addToCartButton = screen.getAllByRole('button')[0];
    const freeTrialButton = screen.getAllByRole('button')[1];

    expect(addToCartButton).not.toBeDisabled();
    expect(freeTrialButton).not.toBeDisabled();
  });

  it('renders settings button only for modules with config', () => {
    render(<List modules={mockModules} disabled={false} onFreeTrial={mockOnFreeTrial} onSettings={mockOnSettings} />);

    mockModules.forEach((module, index) => {
      const buttons = screen.getAllByRole('button');
      const hasSettingsButton = module.config !== null;
      const settingsButtonIndex = index * 3 + 2;

      if (hasSettingsButton) {
        expect(buttons[settingsButtonIndex]).toBeInTheDocument();
      }
    });
  });

  it('handles add to cart for module without config', () => {
    const simpleModule = mockModules[2];
    render(<List modules={[simpleModule]} disabled={false} onFreeTrial={mockOnFreeTrial} onSettings={mockOnSettings} />);

    const addToCartButton = screen.getAllByRole('button')[0];
    fireEvent.click(addToCartButton);

    expect(mockAddModule).toHaveBeenCalledWith({
      id: simpleModule.id,
      name: simpleModule.name,
      price: simpleModule.price,
    });
  });

  it('disables actions when module is in cart', () => {
    (useCart as jest.Mock).mockReturnValue({
      addModule: mockAddModule,
      state: { modules: [{ name: mockModules[0].name }] },
    });

    render(<List modules={[mockModules[0]]} disabled={false} onFreeTrial={mockOnFreeTrial} onSettings={mockOnSettings} />);

    const buttons = screen.getAllByRole('button');
    expect(buttons[0]).toBeDisabled();
    expect(buttons[1]).toBeDisabled();
  });

  it('handles multiSelect config type correctly', () => {
    const moduleWithMultiSelect = mockModules[1];
    render(<List modules={[moduleWithMultiSelect]} disabled={false} onFreeTrial={mockOnFreeTrial} onSettings={mockOnSettings} />);

    const settingsButton = screen.getAllByRole('button')[2];
    fireEvent.click(settingsButton);

    expect(mockOnSettings).toHaveBeenCalledWith(moduleWithMultiSelect);
  });

  it('formats dates correctly', () => {
    render(<List modules={[mockModules[0]]} disabled={false} onFreeTrial={mockOnFreeTrial} onSettings={mockOnSettings} />);

    expect(screen.getByText('2024/01/01')).toBeInTheDocument();
    expect(screen.getByText('2025/01/01')).toBeInTheDocument();
  });
});
