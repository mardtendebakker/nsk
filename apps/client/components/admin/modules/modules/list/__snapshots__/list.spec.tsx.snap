// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component matches snapshot 1`] = `
<DocumentFragment>
  <table
    class="MuiTable-root css-rqglhn-MuiTable-root"
  >
    <thead
      class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
      >
        <td>
          Name
        </td>
        <td>
          Price
        </td>
        <td>
          Active At
        </td>
        <td>
          Expires At
        </td>
        <td>
          Active
        </td>
        <td>
          Actions
        </td>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-hover css-1xe9h37-MuiTableRow-root"
      >
        <td>
          Basic Module
        </td>
        <td>
          € 99,99
        </td>
        <td>
          2024/01/01
        </td>
        <td>
          2025/01/01
        </td>
        <td>
          --
        </td>
        <td>
          <span
            class=""
            data-mui-internal-clone-element="true"
          >
            <button
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
                />
              </svg>
            </button>
          </span>
          <span
            class=""
            data-mui-internal-clone-element="true"
          >
            <button
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PlayArrowIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8 5v14l11-7z"
                />
              </svg>
            </button>
          </span>
          <span
            class=""
            data-mui-internal-clone-element="true"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SettingsIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </span>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root MuiTableRow-hover css-1xe9h37-MuiTableRow-root"
      >
        <td>
          Advanced Module
        </td>
        <td>
          € 149,99
        </td>
        <td>
          2024/02/01
        </td>
        <td />
        <td>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
            />
          </svg>
        </td>
        <td>
          <span
            class=""
            data-mui-internal-clone-element="true"
          >
            <button
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
                />
              </svg>
            </button>
          </span>
          <span
            class=""
            data-mui-internal-clone-element="true"
          >
            <button
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PlayArrowIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8 5v14l11-7z"
                />
              </svg>
            </button>
          </span>
          <span
            class=""
            data-mui-internal-clone-element="true"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SettingsIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </span>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root MuiTableRow-hover css-1xe9h37-MuiTableRow-root"
      >
        <td>
          Simple Module
        </td>
        <td>
          € 49,99
        </td>
        <td />
        <td />
        <td>
          --
        </td>
        <td>
          <span
            class=""
            data-mui-internal-clone-element="true"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </span>
          <span
            class=""
            data-mui-internal-clone-element="true"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PlayArrowIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8 5v14l11-7z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
